core/ingest/src/main/java/mil/nga/giat/geowave/core/ingest/kafka/IngestFromKafkaDriver.java

```diff
@@ -238,88 +218,101 @@ public void consumeFromTopic(
-		topicCount.put(
-				formatPluginName,
-				1);
-
-		final Map<String, List<KafkaStream<byte[], byte[]>>> consumerStreams = consumer.createMessageStreams(topicCount);
-		final List<KafkaStream<byte[], byte[]>> streams = consumerStreams.get(formatPluginName);
-		int counter = 1;
-		for (final KafkaStream stream : streams) {
-			final ConsumerIterator<byte[], byte[]> it = stream.iterator();
-			queue.remove(formatPluginName);
+		try {
 			LOGGER.info("Kafka consumer setup for format [" + formatPluginName + "] against topic [" + formatPluginName + "]");
-			try {
-				while (it.hasNext()) {
-					final byte[] msg = it.next().message();
-					counter++;
-					LOGGER.debug("[" + formatPluginName + "] message received");
-					final I dataRecord = GenericAvroSerializer.deserialize(
-							msg,
-							avroFormatPlugin.getAvroSchema());
-
-					if (dataRecord != null) {
-						try {
-							processMessage(
-									dataRecord,
-									ingestRunData,
-									avroFormatPlugin);
-						}
-						catch (final Exception e) {
-							LOGGER.error("Error processing message: " + e.getMessage());
+			final Map<String, Integer> topicCount = new HashMap<>();
+			topicCount.put(
+					formatPluginName,
+					1);
+
+			final Map<String, List<KafkaStream<byte[], byte[]>>> consumerStreams = consumer.createMessageStreams(topicCount);
+			final List<KafkaStream<byte[], byte[]>> streams = consumerStreams.get(formatPluginName);
+
+			queue.remove(formatPluginName);
+			consumeMessages(
+					formatPluginName,
+					avroFormatPlugin,
+					ingestRunData,
+					streams.get(0));
+		}
+		finally {
+			consumer.shutdown();
+		}
+	}
+
+	protected <T> void consumeMessages(
+			final String formatPluginName,
+			final AvroFormatPlugin<T, ?> avroFormatPlugin,
+			final IngestRunData ingestRunData,
+			final KafkaStream<byte[], byte[]> stream ) {
+		int currentBatchId = 0;
+		try {
+			final ConsumerIterator<byte[], byte[]> messageIterator = stream.iterator();
+			while (messageIterator.hasNext()) {
+				final byte[] msg = messageIterator.next().message();
+				LOGGER.info("[" + formatPluginName + "] message received");
+				final T dataRecord = GenericAvroSerializer.deserialize(
+						msg,
+						avroFormatPlugin.getAvroSchema());
+
+				if (dataRecord != null) {
+					try {
+						processMessage(
+								dataRecord,
+								ingestRunData,
+								avroFormatPlugin);
+						if (++currentBatchId > BATCH_SIZE) {
+							ingestRunData.flush();
+							currentBatchId = 0;
 						}
 					}
+					catch (final Exception e) {
+						LOGGER.error(
+								"Error processing message: " + e.getMessage(),
+								e);
+					}
 				}
 			}
-			catch (final Exception e) {
-
-				LOGGER.warn("Consuming from Kafka topic [" + formatPluginName + "] was interrupted... ");
-
+		}
+		catch (final ConsumerTimeoutException te) {
+			if (kafkaOptions.isFlushAndReconnect()) {
+				LOGGER.info(
+						"Consumer timed out from Kafka topic [" + formatPluginName + "... ",
+						te);
```

Comment:
maybe differentiate this print out to mention that a flush and reconnect will be attempted